timemachine/fe/mcgregor.py

```diff
@@ -360,10 +361,15 @@ def mcs(
 
         total_nodes_visited += mcs_result.nodes_visited
 
+        if total_nodes_visited >= max_node_visits:
```

Comment:
I think you need to do this check *within* the recursion() call to be safe - the number of nodes visited is usually exponential w.r.t. layer - so even going up one layer may result in something like 100x more nodes visited.

Reply:
(i.e. the last iteration of the inner recursion call may visit 10x the `max_node_visits`)

Reply:
Reduces the visits if the per threshold quantity exceeds the total amount in https://github.com/proteneer/timemachine/pull/1357/commits/752469e52eeab9fae783911109df39c3afee70a2